### PR TITLE
[Translation] Fix translator overlapse

### LINKED ISSUE #46189: [Translator] Domain overlapse with intl
### Symfony version(s) affected

4.4

### Description

Injecting domain to translator can induce format errors due to domain mixup

### How to reproduce

With a colision in domain.yml and domain+intl-icu.yml, we got domain.yml data with intl formatter

### Possible Solution

_No response_

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by xavren**
| Q             | A
| ------------- | ---
| Branch?       | 4.4
| Bug fix?      | yes
| New feature?  | no 
| Deprecations? | no 
| Tickets       | Fix #46189 
| License       | MIT
| Doc PR        | symfony/symfony-docs#...

### CODE REVIEW

File: src/Symfony/Component/Translation/Tests/TranslatorTest.php [hunk 1]
```diff
@@ -683,6 +683,20 @@ public function testIntlFormattedDomain()
         $this->assertSame('Hi Bob', $translator->trans('some_message', ['%name%' => 'Bob']));
     }
 
+    /**
+     * @requires extension intl
```
Comment (stof) [inline src/Symfony/Component/Translation/Tests/TranslatorTest.php]: This test does not require the intl extension. The test will also work when using `symfony/polyfill-intl-icu` (and the testsuite always has it available)

Comment (xavren) [reply src/Symfony/Component/Translation/Tests/TranslatorTest.php]: Ok thanks i removed it :)

**Review by nicolas-grekas — CHANGES_REQUESTED** [review]
I'm sorry but this does not make sense to me as MessageCatalog::get() starts with:
```php
        if (isset($this->messages[$domain.self::INTL_DOMAIN_SUFFIX][$id])) {
            return $this->messages[$domain.self::INTL_DOMAIN_SUFFIX][$id];
        }
```

To confirm, I checked out the PR and reverted the patch: tests still pass.
Can you figure out a tests that fails without the patch?

**Comment by nicolas-grekas** [comment]
@xavren still up to finish this one? Can you figure out a failing test case?

**Comment by xavren** [comment]
@nicolas-grekas sorry for the delay, we had some new priorities and forgot to follow this issue.

You are right, this is not the issue, but the bug exists as demonstrated by this https://github.com/xavren/symfony-translator-bug/blob/master/src/Command/TestCommand.php

Code 
```php
$output->writeln($this->translator->trans('test', ['name' => 'World']));
$output->writeln($this->translator->trans('test', ['name' => 'World'], 'messages+intl-icu'));
$output->writeln($this->translator->trans('test', ['name' => 'World'], 'messages'));
```

Result is
```bash
php bin/console app:test
Hello %name%
Hello %World%
Hello %name%
```

It seems that MessageCatalogue loading messages has an issue but cannot spot it and did not find any tests :/

For the debug, here is the cache file

var/cache/dev/translations/catalogue.en.bBuRKVV.php
```php
<?php

use Symfony\Component\Translation\MessageCatalogue;

$catalogue = new MessageCatalogue('en', array (
  'messages+intl-icu' => 
  array (
    'test' => 'Hello %name%',
  ),
));


return $catalogue;
```

**Comment by stof** [comment]
Well, using `%name%` in a catalogue that has the intl suffix is not replaced at all and this is expected, as this is not a placeholder at all in the ICU message format.

**Comment by xavren** [comment]
Completly ok with that, but you can see the content of my translations files 
image

Result is completly mixed up

**Comment by stof** [comment]
@xavren The legacy format replaces whatever you pass as key. so to replace `%name%`, you need to pass `%name%` as key, not `name` (as part if the Intl ICU message format migration, `%` are trimmed from the placeholder name, but they are not _added_ automatically)

**Comment by xavren** [comment]
@stof, the idea was to show mixup catalogue data, we have legacy data in intl-icu domain, that override intl-icu data, formatted by intlFormatter. 
If you tell me this is what's expected, i'll close this issue

**Comment by stof** [comment]
you should not have legacy data in your intl-icu catalogue. That should go in the non-suffixed catalogue for that domain. Putting legacy data in the catalogue that is meant to contain ICU format will indeed not work, and that's totally expected.

**Comment by xavren** [comment]
Hi @stof and @nicolas-grekas, i modified my PR adding tests  that demonstrate the issue (hopefully)

public function testIntlDomainOverlapseWithIntlResourceBefore() does not pass, because MessageCatalogue is mixing up legacy / intl domains

**Review by nicolas-grekas — APPROVED** [review]
This issue was likely introduced in #35430
I fixed it, all green. Thanks for the test case.

**Comment by nicolas-grekas** [comment]
Thank you @xavren.

**Pull request merged**